### PR TITLE
rpm: retry installing packages if mirrors are temporarily unreachable

### PR DESCRIPTION
**Pull request by lxbsz**
Locally I had hit the same issue when installing packages it will fail
with repo metadata downloading and some other failures. Mostly this can
recovery later.

Fixes: https://tracker.ceph.com/issues/47852
Signed-off-by: Xiubo Li <xiubli@redhat.com>

### CODE REVIEW

File: teuthology/task/install/rpm.py [hunk 1]
```diff
@@ -145,16 +145,19 @@ def _downgrade_packages(ctx, remote, pkgs, pkg_version, config):
     remote.run(args='sudo yum -y downgrade {}'.format(' '.join(pkgs_opt)))
     return [pkg for pkg in pkgs if pkg not in downgrade_pkgs]
 
-def _retry_if_503_in_output(remote, args):
+def _retry_if_failures_are_restorable(remote, args):
```
Comment (batrick) [inline teuthology/task/install/rpm.py:148]:
```suggestion
def _retry_if_failures_are_recoverable(remote, args):
```

**Review by batrick — CHANGES_REQUESTED** [review]

File: teuthology/task/install/rpm.py (hunk 1 above)
Comment (lxbsz) [reply teuthology/task/install/rpm.py]: Done, thanks.

**Comment by kshtsk** [comment]
do we still want this? if not, maybe close?

**Comment by lxbsz** [comment]
@batrick ping

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
> do we still want this? if not, maybe close?

Yes

**Comment by batrick** [comment]
@susebot run deploy

**Review by kshtsk — APPROVED** [review]

**Pull request merged**